### PR TITLE
Cloudsmith pull through doc

### PR DESCRIPTION
**Pull request by SharpRake**
## Type of change

This PR adds a new pull-through guide focused on Cloudsmith. It walks readers through setting up a Cloudsmith repo and upstream proxy they can use as a pull through cache for both Developer and Production Images.

I also made some small tweaks to our existing pull through guides.

### What should this PR do?

resolves https://github.com/chainguard-dev/internal/issues/4047

### What are the acceptance criteria? 

n/a

### How should this PR be tested?

This may be slightly annoying to test, as you'll need to set up a Cloudsmith account and start a free trial. Should have a full tech test and a close read through for spelling/grammar/style.

### CODE REVIEW

File: content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+```sh
+docker pull docker.cloudsmith.io/chainguard-example/chainguard-private/chainguard.edu/python:3.9-dev
+```
+
+Once this command is completed you will find the Production Image you pulled in your Cloudsmith repository.
+
+![Screenshot of a portion of a Cloudsmith "Repository" page showing three node packages, each labeled "chainguard.edu/node".](cloudsmith-5.png)
+
+If you run into issues pulling images like this, be sure that your `docker pull` command specifies the correct Cloudsmith organization and repository as well as the correct Chainguard registry.
```
Comment (amouat) [inline content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md:149]: I never got the pull to work, got different errors at various points, but I do see the images in cloudsmith.

File: content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+If you run into issues pulling images like this, be sure that your `docker pull` command specifies the correct Cloudsmith organization and repository as well as the correct Chainguard registry. 
+
+
+## Debugging Pull Through from the Chainguard Registry to Cloudsmith
+
+If you run into issues when trying to pull Images from Chainguard's Registry to Cloudsmith, please make sure the following requirements are met:
+
+* Ensure that all Images [network requirements](https://edu.chainguard.dev/chainguard/administration/network-requirements/) are met.
+* When configuring a remote Cloudsmith repository, ensure that the **URL** field is set correctly. For Developer Images, this should be `https://cgr.dev/chainguard`; for Production Images this should be `https://cgr.dev/`. This field **must not** contain any additional components.
```
Comment (amouat) [inline content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md:157]: Be careful with the final `/` -- you didn't have it in the text above (but it was in the screenshot).

**Review by amouat — APPROVED** [review]
LGTM, comments are all pretty minor.

Good job on the walkthrough and screenshots!

File: content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md (hunk 1 above)
Comment (SharpRake) [reply content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md:149]: TBH I wasn't getting consistent outputs either, which is why I'm not showing any output in this tutorial. 

It "works" though, so I'm putting it on Cloudsmith to fix.

File: content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md (hunk 2 above)
Comment (SharpRake) [reply content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md:157]: amazing catch!

File: content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+```sh
+docker login docker.cloudsmith.io
+```
+
+This command will prompt you to enter your Cloudsmith username and password. Your username appears in the top-right corner of the Cloudsmith web app. If you click on this, a drop-down menu will appear. Select **API Settings**; on the resulting page you'll find a field named **API Key** containing a 40-character string. You can use this API key to access Cloudsmith programmatically or, as in the case of this example, use it as a password in a `docker login` command.
+
+After running the command, you will be able to pull a Chainguard Developer Image through your new Cloudsmith repository. The following example pulls the `nginx` Image:
+
+```sh
+docker pull docker.cloudsmith.io/<cloudsmith-organization>/<cloudsmith-repository>/nginx:latest
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md:76]: Yeah, not great but again I'm inclined to under-explain here.

**Pull request merged**